dotnet/test/common/SeleniumManagerTest.cs

```diff
@@ -0,0 +1,119 @@
+// <copyright file="SeleniumManagerTest.cs" company="Selenium Committers">
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+// </copyright>
+
+using NUnit.Framework;
+using OpenQA.Selenium.Chrome;
+using OpenQA.Selenium.Edge;
+using OpenQA.Selenium.Environment;
+using OpenQA.Selenium.Firefox;
+using OpenQA.Selenium.Safari;
+using System;
+using System.IO;
+
+namespace OpenQA.Selenium;
+
+[TestFixture]
+[Category("SeleniumManager")]
+[IgnoreBrowser(Browser.IE, "IE does not use Selenium Manager")]
+[IgnoreBrowser(Browser.Safari, "Safari does not need Selenium Manager")]
+[IgnoreBrowser(Browser.Remote, "Remote does not use Selenium Manager directly")]
+public class SeleniumManagerTest
+{
+    private static readonly string CacheDirectory = ResolveCacheDirectory();
+
+    private static string ResolveCacheDirectory()
+    {
+        string? cacheDirectory = System.Environment.GetEnvironmentVariable("SE_CACHE_PATH");
+
+        if (!string.IsNullOrWhiteSpace(cacheDirectory))
+        {
+            return cacheDirectory;
+        }
+
+        string homeDirectory = System.Environment.GetFolderPath(System.Environment.SpecialFolder.UserProfile);
+        if (!string.IsNullOrWhiteSpace(homeDirectory))
+        {
+            return Path.Combine(homeDirectory, ".cache", "selenium");
+        }
```

Comment:
This is responsibility of SM, dotnet should not know about this folder.

Reply:
This is a way for us to see that it has in fact been downloaded and not just using the current installation. The contract is SE_CACHE_PATH which we check first. This is currently failing in windows, and again, depending on why would validate the need for this test. 😄

We could also set SE_CACHE_PATH as part of the test which would exercise the contract that way. That might be the right way to do this.